Replace KrMC_pmaps file

gonzaponte opened the pull request:
This PR replaces the previous `KrMC_pmaps` file with a new one. We need it to contain MC tracks (#374) and we've (I've) lost track of the original dataset it has been taken from. Besides the replacement:
- a file containing the output of `dorothea` (with a particular configuration) has been added
- a file containing the output of `penthesilea` (with a particular configuration) has been added
- the tests have been adapted to this new file and improved a bit in some cases
- the hard-coded dependencies of some tests have been fixed
- the fixture that manages the relevant file info has been extended to contain more data
- some "duplicated" tests have been combined

On top of that, I have cleaned up a bit the modules I've touched.

gonzaponte committed:
Replace test file with one containing MC tracks

gonzaponte committed:
Add true output of dorothea under certain conditions

gonzaponte committed:
Add fixture with the conditions and true output of a dorothea run

gonzaponte committed:
Fix test_dorothea_KrMC

The test has been adapted to the new data, but it also
includes a reformulation of the test in terms of the
true output given by the fixture.

gonzaponte committed:
Fix pmaps tests

They contained hard-coded values based on the previous
test file. Besides fixing the test itself, this commit
removes that dependence.

gonzaponte committed:
Combine common tests and remove hard-coded values

This commit:
- Removes hard-coded values from tests
- Rewrites some tests so they don't depend
  on those values or
- Moves that dependence to the fixture
- Combine tests common to df_to_s1_dict,
  df_to_s2_dict and df_to_s2si_dict
- Renames those tests

gonzaponte committed:
PEP8-ify spacing between functions

gonzaponte committed:
Rewrite dst_io_test in terms of the new fixture

gonzaponte committed:
Rewrite kdst_io_test in terms of the new fixture

gonzaponte committed:
Remove unused fixture

gonzaponte committed:
Remove glorious test

gonzaponte committed:
Remove unused imports

gonzaponte committed:
Replace input file in config

gonzaponte committed:
Fix test_config_drive_penthesilea_counters

gonzaponte committed:
Add true output of penthesilea under certain conditions

gonzaponte committed:
Add fixture with the conditions and true output of a penthesilea run

gonzaponte committed:
Reformulate and rename test_penthesilea_run_on_Kr_MC_ipmt_pmaps_5evt

It now uses the new fixture with the true ouput of penthesilea
for a particular configuration. It has been renamed because
there were no ipmt involved.

gonzaponte committed:
Replace star imports with a explicit one

gonzaponte committed:
Update KrMC_pmaps fixture

abotas commented on invisible_cities/reco/pmaps_functions_test.py:
```diff
@@ -70,73 +58,58 @@ def timebin_size_must_be_equal_to_stride_times_25_ns_shows_bug_in_old_data(KrMC_
+    for i, max_events in enumerate(Si_evts):
+        si_dict = df_to_s2si_dict(s2t, sit, max_events)
+        assert sorted(si_dict.keys()) == Si_evts[:i]
+
```
Maybe this could be parameterized?

gonzaponte replied:
Not obvious to me how to do it properly, in a way it is obvious what I am doing. Because it uses particular items in the fixture, it is not something I can set externally.

I can return namedtuples in the fixture, though. That would make the test parametrizable in a readable way.

abotas replied:
Ok it just seemed like repeated code. Not a big deal obviously but ideally i think we would make it a little more concise

gonzaponte replied:
Yes, I didn't like it either, but at the time I decided it was better to repeat the code than to make it obscure. If you like the idea of namedtuples I can implement it tomorrow.

abotas replied:
That sounds good to me

abotas commented on invisible_cities/io/dst_io_test.py:
```diff
@@ -1,37 +1,26 @@
-    (filename, group, node), df = Kr_dst_data
-    dst = load_dst(filename, group, node)
+def test_load_dst(KrMC_kdst):
+    # The function load_dst is actually used in the fixture
```
Maybe `load_dst` should be loaded (again) here even if it slows down the tests a couple hundred microseconds just in case someone changes the fixture?

gonzaponte replied:
Indeed

gonzaponte committed:
Fix single-line imports

gonzaponte committed:
Remove commented-out line

gonzaponte committed:
Return namedtuples in some fixtures

gonzaponte committed:
Use namedtuples in tests

gonzaponte committed:
Reformulate test

The test
timebin_size_must_be_equal_to_stride_times_25_ns_shows_bug_in_old_data
has been reformulated, as it was unreadable.
It also had a hardcoded evt number that, incredibly,
matched one in the input file. It doesn't depend on that, now.

gonzaponte committed:
Perform test_load_dst properly

Use the function to be tested rather than
take advantage of the fixture.

abotas reviewed (approved):
Looks good! And you fixed an ugly old test of mine I think. Wonderful.
